paimon-common/src/main/java/org/apache/paimon/utils/DateTimeUtils.java

```diff
@@ -76,6 +84,95 @@ public class DateTimeUtils {
                     .optionalEnd()
                     .toFormatter();
 
+    public static final String TIMESTAMP_UTC_FLAG = "/UTC";
+
+    public static final String[] AUTO_PARSE_PATTERNS = {
+        "yyyy",
+        "yyyy-MM",
+        "yyyy-MM-dd",
+        "yyyy-MM-dd HH",
+        "yyyy-MM-dd HH:mm",
+        "yyyy-MM-dd HH:mm:ss",
+        "yyyy-MM-dd HH:mm:ss.SSS"
+    };
+    private static final String FULL_TIMESTAMP_DEF = "0000-01-01 00:00:00.000";
+
+    /**
+     * auto format date.
+     *
+     * @param timestampString
+     * @return
+     */
+    public static Timestamp autoFormatToTimestamp(String timestampString) {
```

Comment:
Can we just use `parseTimestampData`? It is very confused to introduce another parser here.

Reply:
@JingsongLi  Fixed，Please recheck!